zulipterminal/ui_tools/boxes.py

```diff
@@ -898,22 +898,29 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     else:
                         header.focus_col = self.FOCUS_HEADER_BOX_STREAM
                 else:
-                    all_valid = self._tidy_valid_recipients_and_notify_invalid_ones(
-                        self.to_write_box
-                    )
-                    if not all_valid:
-                        return key
-                    # We extract recipients' user_ids and emails only once we know
-                    # that all the recipients are valid, to avoid including any
-                    # invalid ones.
-                    self.update_recipients(self.to_write_box)
+                    if isinstance(self.to_write_box, ReadlineEdit):
```

Comment:
Can we use the same conditional here as below? I expect isinstance to work, but it's not as descriptive as the current state.

Reply:
If I have understood correctly you are referring to `msg_edit_state`. Yep, that should be a better choice as it seems more basic. Probably used `isinstance(self.to_write_box, ReadlineEdit)` because I was thinking along the lines of `to_write_box` being editable or not as the variable between two cases.

Reply:
Right, `msg_edit_state` is used in the logic below.